aggregation/simple/src/main/java/com/spotify/heroic/aggregation/simple/FilterPointsKThresholdStrategy.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2015 Spotify AB.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.heroic.aggregation.simple;
+
+import com.spotify.heroic.metric.*;
+import lombok.Data;
+
+import java.util.*;
+
+import static java.util.stream.Collectors.toList;
+
+@Data
+public class FilterPointsKThresholdStrategy implements MetricMappingStrategy{
+    private final FilterKThresholdType filterType;
+    private final double k;
+
+    @Override
+    public MetricCollection apply(MetricCollection metrics) {
+        return MetricCollection.build(
+            MetricType.POINT,
+            filterWithThreshold(metrics.getDataAs(Point.class))
```

Comment:
This would cause a runtime exception if the data type is not `Point`. Better make sure that all the available types can be handled or that the aggregation ignores non-point data.